src/file_tree.ml

```diff
@@ -65,8 +65,17 @@ let load ?(extra_ignored_subtrees=Path.Set.empty) path =
     let files = String_set.of_list files in
     let ignored_sub_dirs =
       if not ignored && String_set.mem "jbuild-ignore" files then
-        String_set.of_list
-          (Io.lines_of_file (Path.to_string (Path.relative path "jbuild-ignore")))
+        let ignore_file = Path.to_string (Path.relative path "jbuild-ignore") in
+        let files =
+          Io.lines_of_file ignore_file
+        in
+        let remove_subdirs index fn =
+          (Filename.dirname fn = Filename.current_dir_name) ||
+            (Loc.(warn (of_pos (ignore_file, index + 1, 0, String.length fn))
+            "subdirectory expression %s ignored" fn);
+             false)
```

Comment:
I think a good old `if ... then ... else ...` would be clearer here. You must be doing too much shell programming :p

Reply:
Well, maybe a bit, but it's more that I don't like `if p then true else (thing; false)` so much!